bindings/pyroot_experimental/PyROOT/src/TDirectoryFilePyz.cxx

```diff
@@ -0,0 +1,71 @@
+// Author: Massimiliano Galli CERN  08/2019
+// Original PyROOT code by Wim Lavrijsen, LBL
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "CPyCppyy.h"
+#include "CPPInstance.h"
+#include "CPPOverload.h"
+#include "PyROOTPythonize.h"
+#include "ProxyWrappers.h"
+#include "Python.h"
+#include "Utility.h"
+#include "PyzCppHelpers.hxx"
+#include "TClass.h"
+#include "TKey.h"
+#include "TDirectoryFile.h"
+
+using namespace CPyCppyy;
+
+////////////////////////////////////////////////////////////////////////////
+/// \brief Allow access to objects through the method Get()
+/// This concerns both TDirectoryFile and TFile, since the latter
+/// inherits the Get method from the former.
+/// We decided not to inject this behavior directly in TDirectory in order
+/// to keep the same implementation of the older version of PyROOT.
+/// In case the object is not found, a null pointer is returned.
+PyObject *TDirectoryFileGetPyz(const CPPInstance *self, PyObject *pynamecycle)
+{
+   // Pythonization of TDirectoryFile::Get that handles non-TObject deriveds
+   if (!CPPInstance_Check(self)) {
+      PyErr_SetString(PyExc_TypeError,
+                      "T(Directory)File::Get must be called with a T(Directory)File instance as first argument");
+      return nullptr;
+   }
+   auto dirf = (TDirectoryFile *)GetTClass(self)->DynamicCast(TDirectoryFile::Class(), self->GetObject());
+   if (!dirf) {
+      PyErr_SetString(PyExc_ReferenceError, "attempt to access a null-pointer");
+      return nullptr;
+   }
+   const char *namecycle = CPyCppyy_PyUnicode_AsString(pynamecycle);
+   if (!namecycle)
+      return nullptr; // TypeError already set
+   auto key = dirf->GetKey(namecycle);
+   if (key) {
+      void *addr = dirf->GetObjectChecked(namecycle, key->GetClassName());
+      return BindCppObjectNoCast(addr, (Cppyy::TCppType_t)Cppyy::GetScope(key->GetClassName()), kFALSE);
+   }
+   void *addr = dirf->Get(namecycle);
```

Comment:
Wasn't this the code that was executed only for `TDirectory`, according to the comment it had above?
You kept it because it is present in the old PyROOT, which just calls the normal `Get` if it can't find the key?

Reply:
exactly

Reply:
Ok, can you rescue the comment it has in the old PyROOT and put it here too?

Reply:
sure